### DISCUSSION TITLE
Question about adj values in cluster.py

### DISCUSSION THREAD

**Discussion by Antoine-ls** [pyg-team/pytorch_geometric#2779] [Q&A] [answered]
I using ClusterGCN for node classification, and I want to get degree of one node in the adjacent matrix, however, I find that the adj is defined in cluster.py as follows, I do not understand why you set value to torch.arange(E, device=data.edge_index.device)  instead of torch.ones(E), I want to calculate degree by adj.sum(0) but I can not do this in this adj, I wonder if this definition is necessary or if you have any suggestions for me to calculate degree in this sparse adj?

adj = SparseTensor(
                row=data.edge_index[0], col=data.edge_index[1],
                value=torch.arange(E, device=data.edge_index.device),
                sparse_sizes=(N, N))

**Comment by rusty1s** [comment, marked as answer]
`adj` is just an internal storage of the `ClusterData`, which means that you probably shouldn't operate on it. We save a `torch.arange` vector in there as value in order to determine the original ID of edges when creating subgraphs (in order to filter edge features).

If you want to get the node degree of nodes in your graph, you can utilize `torch_geometric.utils.degree`, either on the original `edge_index`, or on the subgraph data `data.edge_index` returned by the `ClusterData`/`ClusterLoader`.

**Reply by Antoine-ls** [reply]
Understood, thanks for your reply!
